contracts/multiswap-contracts/FiberRouter.sol

```diff
@@ -592,29 +670,45 @@ function _swapOnSameNetwork(
      * @param salt The salt for unique tx 
      * @param expiry The expiration time for the signature
      * @param multiSignature The multisig validator signature
-    */
+     * @param cctpType Type of withdrawal: true for CCTP, false for normal
+     */
     function withdrawSigned(
         address token,
         address payee,
         uint256 amount,
         bytes32 salt,
         uint256 expiry,
-        bytes memory multiSignature
+        bytes memory multiSignature,
+        bool cctpType
     ) public virtual nonReentrant {
-        // Validation checks
+        // Validate input parameters
         require(token != address(0), "FR: Token address cannot be zero");
-        require(payee != address(0), "Payee address cannot be zero");
-        require(amount != 0, "Amount must be greater than zero");
-        require(salt > bytes32(0), "salt must be greater than zero bytes");
-        // need to add restrictions
-        amount = FundManager(pool).withdrawSigned(
-            token,
-            payee,
-            amount,
-            salt,
-            expiry,
-            multiSignature
-        );
+        require(payee != address(0), "FR: Payee address cannot be zero");
+        require(amount != 0, "FR: Amount must be greater than zero");
+        require(salt > bytes32(0), "FR: Salt must be greater than zero bytes");
+
+        if (cctpType) {
```

Comment:
Cctp withdrawal is done when we have we need to withdraw funds from cctp fund manager to perform a bigger amount of usdc swap such as 50K USDC. When we perform cctp swap, we deposit our usdc funds to target cctp fund manager using cctp and on target network, we can call cctpFundManager's withdraw function to get the amount moved to target user address. 

We can create separate functions but that will increase the size of contract as in both cases we are doing the same checks and actions. However, in one function all we need to do is to check off swap type (cctp type == true) then we can call cctp FundManager withdraw function otherwise normal flow will be performed.